zone/client.cpp

```diff
@@ -7455,6 +7455,13 @@ void Client::ProcessXTargetAutoHaters()
 
 		if (XTargets[i].ID != 0 && !GetXTargetAutoMgr()->contains_mob(XTargets[i].ID)) {
 			XTargets[i].ID = 0;
+			XTargets[i].Name[0] = 0;
+			XTargets[i].dirty = true;
+		}
+
+		if (XTargets[i].ID != 0 && entity_list.GetMob(XTargets[i].ID) && !entity_list.GetMob(XTargets[i].ID)->IsValidXTarget()) {
```

Comment:
Two entity list calls is a bit expensive here. Should be fetching the mob once, checking if the mob pointer is valid and then accessing it. We are doing two entity list searches to fetch the same entity